library_generation/test/generate_library_unit_tests.sh

```diff
@@ -181,15 +181,14 @@ generate_library_failed_with_invalid_generator_version() {
   local destination="google-cloud-alloydb-v1-java"
   local res=0
   cd "${script_dir}/resources"
-  $("${script_dir}"/../generate_library.sh \
+  "${script_dir}"/../generate_library.sh \
```

Comment:
The reason I wrap this command into `$()` is because `generate_library.sh` will exit if the version is incorrect and use `$()` to force back to the current shell and execute command after `\\`.

I'm not sure whether this is the best approach though, maybe worth an investigation later.

Reply:
Not sure if this approach is good either but I prepended `bash $script` in https://github.com/googleapis/sdk-platform-java/pull/1987/commits/7fc632a0928a225a9c560eaa6eb35b25fbf14057 to prevent the whole test suite from exiting if that's what you mean by `force back to the current shell`. I was having issues when running the unit tests after the adding the OS detection (or a few commits later) function which happened to be fixed by removing `$()`.